host/traceh.c

```diff
@@ -331,6 +348,18 @@ void oe_log_message(bool is_enclave, oe_log_level_t level, const char* message)
     {
         initialize_log_config();
     }
+
+    if (_oe_log_callback)
+    {
+        if (oe_mutex_lock(&_log_lock) == OE_OK)
```

Comment:
Can we try avoiding the lock for each message that is printed?

Reply:
I think the lock can guarantee the sequence of the messages in destination. Although there are timestamps, if the messages are messed up then the log is not readable. Yes we need a better method to avoid performance lost.

Reply:
It would be good to define what the contract is. Are we using the lock to serialize all the log messages or it it being used for thread-safety?

Reply:
Yes serialization may be a side effect of the lock for thread-safety.